va/va.go

```diff
@@ -1141,6 +1145,20 @@ func (va *ValidationAuthorityImpl) PerformValidation(ctx context.Context, domain
 
 	va.log.AuditObject("Validation result", logEvent)
 	va.log.Infof("Validations: %+v", authz)
+
+	// Try to marshal the validation results and prob (if any) to protocol
+	// buffers. We do this later in the VA wrapper but we also want to try this
+	// proactively in the VA such that we can log anytime there is a failure to
+	// marshal. This is typically the result of invalid UTF-8 in user supplied
+	// data that we intended to escape with `replaceInvalidUTF8` but may have
+	// missed. In such a case logging explicitly at this layer in addition to
```

Comment:
You can make this a little clearer: We log at this layer instead of leaving it up to gRPC because gRPC doesn't log the actual contents, making it hard to figure out what's broken.